src/kakarot/instructions.cairo

```diff
@@ -34,80 +34,579 @@ from kakarot.instructions.sha3 import Sha3
 // @custom:namespace EVMInstructions
 namespace EVMInstructions {
     // @notice Decode the current opcode and execute associated function.
-    // @dev The function iterates through the provided instructions and executes each of them
-    //      whilst also performing safety checks and updating the pc counter after each instruction execution
-    // @param instructions The instruction set.
+    // @dev The function uses an internal jump table to execute the corresponding opcode
     // @param ctx The pointer to the execution context.
     // @return The pointer to the updated execution context.
-
     func decode_and_execute{
         syscall_ptr: felt*,
         pedersen_ptr: HashBuiltin*,
-        range_check_ptr,
+        range_check_ptr: felt,
```

Comment:
I've never seen this, what does this do?:)

Reply:
it's the same, I think that I changed this for an easier copy/pasting, which led to the useless cast at line 590 😇